### PR TITLE
Magic folder integration tests

### PR DESCRIPTION
**Pull request by meejah**
This ports all the magic-folders integration-style tests (from `check_magicfolder_smoke.py`) to a proper event-runner, etc. Uses `py.test` fixtures to build up the introducer, storage servers and alice + bob clients. It also runs a `flog_tool` gather instance to grab all logs from all the running services and dumps them at the end of the run.

### CODE REVIEW

File: integration/conftest.py [hunk 1]
```diff
@@ -0,0 +1,511 @@
...
+            reactor, request, temp_dir, tahoe_binary, introducer_furl, flog_gatherer, "alice",
+            web_port="tcp:9980:interface=localhost",
+            storage=False,
+#            magic_text='Completed initial Magic Folder scan successfully',
+        )
+    )
+    try:
+        mkdir(join(temp_dir, 'magic-alice'))
+    except OSError:
+        pass
```
Comment (daira) [inline integration/conftest.py]: Put this before creating the process.

File: integration/conftest.py [hunk 2]
```diff
@@ -0,0 +1,511 @@
...
+
+@pytest.fixture(scope='session')
+def alice(reactor, temp_dir, tahoe_binary, introducer_furl, flog_gatherer, storage_nodes, request):
+    import time ; time.sleep(1)
+    process = pytest.blockon(
+        _create_node(
+            reactor, request, temp_dir, tahoe_binary, introducer_furl, flog_gatherer, "alice",
+            web_port="tcp:9980:interface=localhost",
+            storage=False,
+#            magic_text='Completed initial Magic Folder scan successfully',
```
Comment (daira) [inline integration/conftest.py]: This commented-out thing is kind of misleading.

File: integration/conftest.py [hunk 3]
```diff
@@ -0,0 +1,511 @@
...
+    # point ...
+
+    # XXX hmm, maybe we can just nuke alice's process and it'll get
+    # re-started next time we ask for "alice" fixture...?
+
+    # hmm, looks like we're still "sometimes"/often not waiting long
+    # enough until the grid is really-ready .. maybe re-try this
+    # command a couple times automagically? or figureout what's up w/
+    # the storage nodes?
+    import time ; time.sleep(5)
```
Comment (daira) [inline integration/conftest.py]: Eek

File: integration/test_aaa_aardvark.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# XXX FIXME just putting these here as a cheap hack at getting some
```
Comment (daira) [inline integration/test_aaa_aardvark.py]: I'm fine with this, remove the XXX FIXME.

**Comment by meejah** [comment]
Note to self: try on windows-vm to see if the appveyor failure is "real" or not.

**Comment by warner** [comment]
I've kicked appveyor to re-build that one, maybe it was something transient.

**Comment by meejah** [comment]
Okay, I've addressed the comments and this is ready to go. Note that two of the tests will fail until the PR for 2814 is landed (if @warner agrees I can squash before merge if you want, perhaps after 2814 is landed so I can rebase it as well).

**Comment by warner** [comment]
looks reasonable to me, squash at will

**Comment by meejah** [comment]
Okay, I rebased to master and just squashed everything into one commit.

**Pull request merged**